src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBuildWrapperTest.java

```diff
@@ -0,0 +1,55 @@
+package com.cloudbees.jenkins.plugins.sshagent;
+
+import hudson.model.FreeStyleProject;
+import hudson.tasks.Shell;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import com.cloudbees.jenkins.plugins.sshcredentials.SSHUserPrivateKey;
+import com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+
+public class SSHAgentBuildWrapperTest extends SSHAgentBase {
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    @Before
+    public void starting () throws Exception {
+        startMockSSHServer();
+    }
+
+    @Test
+    public void sshAgentAvailable() throws Exception {
+        List<String> credentialIds = new ArrayList<String>();
+        credentialIds.add("84822271-02d5-47b8-b8ff-c40fef175c29");
+
+        SSHUserPrivateKey key = new BasicSSHUserPrivateKey(CredentialsScope.GLOBAL, credentialIds.get(0), "cloudbees",
+                new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource(getPrivateKey()), "cloudbees", "test");
+        SystemCredentialsProvider.getInstance().getCredentials().add(key);
+        SystemCredentialsProvider.getInstance().save();
```

Comment:
Probably works, but I have never seen this idiom before. Normally I see

```java
CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), c);
```

and (searching GitHub) in one place I saw

```java
SystemCredentialsProvider.getInstance().setDomainCredentialsMap(Collections.singletonMap(Domain.global(), Arrays.<Credentials>asList(c)));
```

Not sure what is the official idiom; @stephenc advice? A helper function for adding credentials to a given target (`Jenkins`, `Folder`, `User`) would certainly be appreciated. (Would be used mainly from functional tests, but also from migration code for plugins originally using a `Secret` or the like.)